src/main.rs

```diff
@@ -1,4 +1,3 @@
-#![deny(warnings)]
```

Comment:
Is it because of all the deprecation warnings around the DMA?

Reply:
NB: I'm fine with getting rid of the `deny(warnings)`. We should aim to get rid of warnings regularly but they should not make development unnecessary hard. In almost all cases they are indeed "only" warnings.

Reply:
Yes, this is due to the DMA deprecation, which is going to be revamped in a follow-on PR. I'm unsure of what the right way to proceed here is because I think generally `[deny(warnings)]` is what we want, but not in this case.

Reply:
Could we add `#![allow(deprecated)]` with a note for why? This would allow us to keep `deny(warnings)` if we wanted, but at least it would prevent other warnings (there are none at the moment) from hiding among the 34 deprecated warnings the DMA code is generating.